sidechain/consensus/common/src/block_import_confirmation_handler.rs

```diff
@@ -105,19 +106,25 @@ impl<
 			.map_err(|e| Error::Other(e.into()))?
 			.map_err(|e| Error::Other(format!("{:?}", e).into()))?;
 
-		let opaque_call =
-			OpaqueCall::from_tuple(&(call, shard, header.block_number(), header.hash()));
+		if header.block_number() == header.next_finalization_block_number() {
+			let opaque_call = OpaqueCall::from_tuple(&(
+				call,
+				shard,
+				header.block_number(),
+				header.next_finalization_block_number() + BLOCK_NUMBER_FINALIZATION_DIFF,
```

Comment:
If I understand the code correctly, you add the if twice now. In the composer and in the importer. I think it should only be called in the composer?

Or am I understanding something wrong?

Reply:
Ah, this would explain why block number 1, still works.

Reply:
see comment above